docs/enhancements/metricsConfig_redesign.md

```diff
@@ -0,0 +1,119 @@
+
+#### minimumDuration
+Deprecate the current implementation of hiveConfig.spec.metricsConfig.metricsWithDuration, and change it to be reported as metricsConfig.clusterDeploymentRelatedMetrics[$name].minimumDuration. The implementation of using the duration as a threshold before we report the metric stays the same.
+
+#### additionalClusterDeploymentLabels
+Deprecate current implementation of hiveConfig.spec.metricsConfig.additionalClusterDeploymentLabels and change it to be reported as metricsConfig.clusterDeploymentRelatedMetrics[$name].additionalClusterDeploymentLabels. It's implementation will not change.
+
+#### clusterDeploymentLabelSelector
+This would be a new feature, of type [LabelSelector](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#LabelSelector), and we'd use the LabelSelector.MatchLabels and/or LabelSelector.MatchExpressions to match the conditions in order to decide if a metric should be reported.
+This encapsulates slightly more advanced filter logic over the existing clusterdeployment labels. In the example above, hive_foo_counter metric will only be reported for the cluster deployments that are labelled with aro-snowflake, not in limited support, and has the metric value > 10m.
+
+### Implementation Details / Notes
+- In order to implement this change, we will have to deprecate the existing hiveconfig.spec.metricsConfig.metricsWithDuration and hiveconfig.spec.metricsConfig.additionalClusterDeploymentLabels.`
+- Current configuration of using a shorthand camelCase key to refer to a corresponding metric while configuring hiveconfig.spec.metricsConfig.metricsWithDuration will be removed, and users will have to use the full metric name in the proposed configuration.
+- Additionally, we would also want to remove `cluster_type` as a fixed label for all the metrics that do report it now. This would be a breaking change as the consumers who want `cluster_type` will have to configure it via metricsConfig.clusterDeploymentRelatedMetrics[].additionalClusterDeploymentLabels for all the metrics they need it for.
+- There are other fixed labels (like platform and version), which are directed fed from the clusterdeployment labels. Ideally we would want to feed them via hiveconfig, but we might need an exception for the cluster deployment labels that are applied by hive. 
+- All the options configured for a metric within clusterDeploymentRelatedMetrics, will work in tandem with each other. For ex, if all possible options are specified for a metric, then that metric will be reported with the additional labels as per additionalClusterDeploymentLabels, and will be reported only if it matches the labels and/or expressions as per clusterDeploymentLabelSelector and if the duration to be reported exceeds the minimumDuration
+- Failure modes:
+  These situations will cause a panic().
+  - clusterDeploymentRelatedMetrics[].name doesn't exist as a metric
```

Comment:
I'm a little leery of this. It makes it hard e.g. for a consumer to share a hiveconfig across heterogeneous deployments with different versions of hive (think ACM).

I do feel like we discussed this, but I can't remember what we landed on, or why. Did we talk about having status conditions instead?

If I'm retreading ground here, it means the reasoning isn't obvious, which means we should have the alternatives discussed in the doc. (Inline or a whole separate "Alternatives" section -- whatever you prefer.)

Reply:
Uh, I don't understand this comment well enough. I personally think it is better to only refer to the full-name of metric in the hiveConfig. It has to be the hiveConfig because we register metrics way before we parse the conditions of hiveOperator - and registration requires us to know all the labels a metric will report. We can only register a metric once during a pod's lifetime. That's why hiveconfig seemed as a better option, and we were okay with panics (because these panics happen within seconds - at most 2 minutes of an hiveConfig edit).
We can consider manifests or volumeMounts or something - the pod has to restart anyways. Do you have any alternatives?

Reply:
Sorry, my comment was around whether we should actually panic when we encounter an unrecognized metric name in hiveconfig.

I don't claim to remember our discussion from last year perfectly... but maybe we decided panic was the lesser evil. For instance, it would catch a typo which might otherwise go unnoticed until too late. Worth the tradeoff of having to make sure you're only referencing metrics available on the hive version running on _this_ hub.

So, I'm just asking that that rationale be explained a little bit. A sentence or two is plenty.